src/layers/data_type_layer.cpp

```diff
@@ -781,28 +812,24 @@ void data_type_layer<InputTensorDataType, OutputTensorDataType>::
 fp_setup_inputs(El::Int mini_batch_size) {
   if (get_num_parents() < 1) { return; }
 
-  // Determine distributed matrix alignment
-  const auto& alignment_dist = get_parent_layer().get_activations(*this).DistData();
-
   // Iterate through input tensors
   for (int i = 0; i < get_num_parents(); ++i) {
+
 #ifdef LBANN_HAS_DISTCONV
+    // Skip if tensors are managed by Distconv
     if (!keep_original_inputs(i)) continue;
 #endif // LBANN_HAS_DISTCONV
+
     // Initialize input tensor
     const auto& parent = get_parent_layer(i);
     const auto& parent_output = parent.get_activations(*this);
     auto& input = *m_inputs[i];
     input.Empty(false);
-    if(!this->is_subgraph_parallelism_enabled()){
-      input.AlignWith(alignment_dist);
```

Comment:
Is this done somewhere else or skipped now?

Reply:
Yep, non-aligned matrices will now trigger communication.